### PR TITLE
fix: show correct asset name for staking info

### PR DESCRIPTION
**Pull request by 0xdef1cafe**
## Description

this was hardcoded to osmo - this now uses a selector to get the correct asset name from the assetId param

## Notice

- [ ] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/CONTRIBUTING.md) guide?
- [ ] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

n/a - found in release testing

## Risk

low - asset name should appear correctly

## Testing

follow the same steps used to discover this bug

### CODE REVIEW

**Review by gomesalexandre — APPROVED** [review]
Nice catch, tested against release branch and this fixes the hardcoded asset name.

**Review by cjthompson — APPROVED** [review]

File: src/state/slices/assetsSlice/selectors.ts [hunk 1]
```diff
@@ -22,8 +22,9 @@ export const selectAssetById = createCachedSelector(
   (byId, assetId) => byId[assetId] || undefined,
 )((_state: ReduxState, assetId: AssetId | undefined): AssetId => assetId ?? 'undefined')
 
-export const selectAssetNameById = createSelector(selectAssetById, asset =>
-  asset ? asset.name : undefined,
+export const selectAssetNameById = createSelector(
+  selectAssetById,
+  (asset): string => asset?.name ?? '',
```
Comment (cjthompson) [inline src/state/slices/assetsSlice/selectors.ts:27]: I don't understand why this change was needed.

Comment (0xdef1cafe) [reply src/state/slices/assetsSlice/selectors.ts:27]: changes the type signature of the function to be safer and more explicit - it was previously returning `string | undefined` and requiring view layer consumers to opt chain on the result. 

the selector should return a string. 

the selector is not used anywhere else in the code base, and was previously completely unused, so there is no risk of regressions.

**Comment by 0xdef1cafe** [comment]
merging as this passes CI locally and will pass CI on the release branch before going to prod
